api/maven-api-core/src/main/java/org/apache/maven/api/ExtensibleEnums.java

```diff
@@ -24,28 +24,69 @@
 import java.util.Objects;
 import java.util.Set;
 
+/**
```

Comment:
Not for this PR, but I wonder if we should have a different name here to avoid confusion with Java enums, which these aren't